packages/socket-mode/src/SlackWebSocket.ts

```diff
@@ -111,29 +111,28 @@ export class SlackWebSocket {
       agent: this.options.httpAgent,
     };
 
-    const ws = new WebSocket(this.options.url, options);
+    this.websocket = new WebSocket(this.options.url, options);
 
-    ws.addEventListener('open', (_event) => {
+    this.websocket.addEventListener('open', (_event) => {
       this.logger.debug('WebSocket open event received (connection established)!');
-      this.websocket = ws;
```

Comment:
Not tracking the current websocket instance was a problem in the case the web socket server returns an error. Since only once the `open` event was emitted would we keep track of the websocket (this line), any errors during the websocket handshake would call into `disconnect` which, crucially, only terminates the underlying network resources for the tracked `this.websocket`. That was the underlying issue that caused the "doubling up" of reconnection attempts.

Reply:
Thank you for finding this. I was able to patch this in and test it while we await a full fix.